tests/flytekit/unit/core/test_node_creation.py

```diff
@@ -427,6 +427,7 @@ def my_wf(a: str) -> str:
     wf_spec = get_serializable(OrderedDict(), serialization_settings, my_wf)
     assert len(wf_spec.template.nodes) == 1
     assert wf_spec.template.nodes[0].metadata.name == "foo"
+    assert wf_spec.template.nodes[0].id == "t-1"
```

Comment:
why dash instead of underscore?

Reply:
underscore can't be used in pod name. https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy